asdf/asdf.py

```diff
@@ -876,7 +876,23 @@ def _write_tree(self, tree, fd, pad_blocks):
         fd.write(b'\n')
 
         if len(tree):
-            yamlutil.dump_tree(tree, fd, self)
+            serialization_context = self._create_serialization_context()
+
+            def _tree_finalizer(tagged_tree):
```

Comment:
A minimal docstring here for `_tree_finalizer()` would be useful to indicate what is happening.